lib/gen_stage/streamer.ex

```diff
@@ -3,35 +3,47 @@ defmodule GenStage.Streamer do
   use GenStage
 
   def start_link({_, opts} = pair) do
-    GenStage.start_link(__MODULE__, pair, opts)
+    {:current_stacktrace, [_info_call | stack]} = Process.info(self(), :current_stacktrace)
+    GenStage.start_link(__MODULE__, {pair, stack}, opts)
   end
 
-  def init({stream, opts}) do
+  def init({{stream, opts}, stack}) do
     continuation =
       &Enumerable.reduce(stream, &1, fn
         x, {acc, 1} -> {:suspend, {[x | acc], 0}}
         x, {acc, counter} -> {:cont, {[x | acc], counter - 1}}
       end)
 
-    {:producer, continuation, Keyword.take(opts, [:dispatcher, :demand])}
+    {:producer, {stack, continuation}, Keyword.take(opts, [:dispatcher, :demand])}
+  end
+
+  def init({stream, opts}) do
+    {:current_stacktrace, [_info_call | stack]} = Process.info(self(), :current_stacktrace)
```

Comment:
`GenStage.from_enumerable` [calls into this directly](https://github.com/elixir-lang/gen_stage/blob/master/lib/gen_stage.ex#L1609). I didn't want to put this information into there because it seemed like the calling function shouldn't be aware of the needed stack trace.

Reply:
Since it is all private API, so we can either add it there or add a from_enumerable function to this module that `from_enumerable` would ccall. Getting the stacktrace from within `init` won't hold any relevant info.

Reply:
Ping. :)